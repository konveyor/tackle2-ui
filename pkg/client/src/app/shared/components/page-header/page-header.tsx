import React from "react";
import {
  Stack,
  StackItem,
  Split,
  SplitItem,
  TextContent,
  Text,
} from "@patternfly/react-core";
import { BreadCrumbPath } from "../breadcrumb-path";
import { MenuActions } from "../menu-actions";
import { HorizontalNav } from "../horizontal-nav/horizontal-nav";

export interface PageHeaderProps {
  title: string;
  description?: React.ReactNode;
  breadcrumbs: { title: string; path: string | (() => void) }[];
  btnActions?: React.ReactNode;
  menuActions: { label: string; callback: () => void }[];
  navItems?: { title: string; path: string }[];
}

export const PageHeader: React.FC<PageHeaderProps> = ({
  title,
  description,
  breadcrumbs,
  btnActions,
  menuActions,
  navItems,
}) => {
  return (
    <Stack hasGutter>
      <StackItem>
        {breadcrumbs.length > 0 && <BreadCrumbPath breadcrumbs={breadcrumbs} />}
      </StackItem>
      <StackItem>
        <Split>
          <SplitItem isFilled>
            <TextContent>
              <Text component="h1">{title}</Text>
              {description}
            </TextContent>
          </SplitItem>
          {btnActions && <SplitItem>{btnActions}</SplitItem>}
          {menuActions.length > 0 && (
            <SplitItem>
              <MenuActions actions={menuActions} />
            </SplitItem>
          )}
        </Split>
      </StackItem>
      {navItems && (
        <StackItem>
          <HorizontalNav navItems={navItems} />
        </StackItem>
      )}
    </Stack>
  );
};
