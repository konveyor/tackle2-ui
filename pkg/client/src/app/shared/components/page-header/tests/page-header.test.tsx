import React from "react";
import { shallow } from "enzyme";
import { PageHeader } from "../page-header";
import { Button } from "@patternfly/react-core";

describe("PageHeader", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(
      <PageHeader
        title="mycompany"
        breadcrumbs={[
          {
            title: "Companies",
            path: "/companies",
          },
          {
            title: "Company details",
            path: "/companies/1",
          },
        ]}
        btnActions={<Button>send email</Button>}
        menuActions={[
          {
            label: "Edit",
            callback: jest.fn(),
          },
          {
            label: "Delete",
            callback: jest.fn(),
          },
        ]}
        navItems={[
          {
            title: "Overview",
            path: "/companies/1/overview",
          },
          {
            title: "YAML",
            path: "/companies/1/yaml",
          },
          {
            title: "SUNAT",
            path: "/companies/1/sunat",
          },
        ]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
