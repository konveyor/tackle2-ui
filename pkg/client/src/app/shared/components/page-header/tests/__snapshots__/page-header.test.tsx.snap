// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Renders without crashing 1`] = `
<Stack
  hasGutter={true}
>
  <StackItem>
    <Component
      breadcrumbs={
        Array [
          Object {
            "path": "/companies",
            "title": "Companies",
          },
          Object {
            "path": "/companies/1",
            "title": "Company details",
          },
        ]
      }
    />
  </StackItem>
  <StackItem>
    <Split>
      <SplitItem
        isFilled={true}
      >
        <TextContent>
          <Text
            component="h1"
          >
            mycompany
          </Text>
        </TextContent>
      </SplitItem>
      <SplitItem>
        <Button>
          send email
        </Button>
      </SplitItem>
      <SplitItem>
        <Component
          actions={
            Array [
              Object {
                "callback": [MockFunction],
                "label": "Edit",
              },
              Object {
                "callback": [MockFunction],
                "label": "Delete",
              },
            ]
          }
        />
      </SplitItem>
    </Split>
  </StackItem>
  <StackItem>
    <Component
      navItems={
        Array [
          Object {
            "path": "/companies/1/overview",
            "title": "Overview",
          },
          Object {
            "path": "/companies/1/yaml",
            "title": "YAML",
          },
          Object {
            "path": "/companies/1/sunat",
            "title": "SUNAT",
          },
        ]
      }
    />
  </StackItem>
</Stack>
`;
